climada/util/coordinates.py

```diff
@@ -1155,23 +1158,81 @@ def get_admin1_info(country_names):
         Shape according to Natural Earth.
     """
 
-    if isinstance(country_names, str):
-        country_names = [country_names]
+    if isinstance(countries, (str, int, float)):
+        countries = [countries]
+    if not isinstance(countries, list):
+        raise TypeError("Invalid type for input parameter 'countries'")
     admin1_file = shapereader.natural_earth(resolution='10m',
                                             category='cultural',
                                             name='admin_1_states_provinces')
     admin1_recs = shapefile.Reader(admin1_file)
     admin1_info = dict()
     admin1_shapes = dict()
-    for iso3 in country_names:
-        admin1_info[iso3] = list()
-        admin1_shapes[iso3] = list()
+    for country in countries:
+        if isinstance(country, (int, float)):
+            country = f'{int(country):03d}' # transform numerric code to str
+        country = pycountry.countries.lookup(country).alpha_3 # get iso3a code
+        admin1_info[country] = list()
+        admin1_shapes[country] = list()
         for rec, rec_shp in zip(admin1_recs.records(), admin1_recs.shapes()):
-            if rec['adm0_a3'] == iso3:
-                admin1_info[iso3].append(rec)
-                admin1_shapes[iso3].append(rec_shp)
+            if rec['adm0_a3'] == country:
+                admin1_info[country].append(rec)
+                admin1_shapes[country].append(rec_shp)
     return admin1_info, admin1_shapes
 
+def get_admin1_geometries(countries):
```

Comment:
Repeating myself: it makes sense to add a `resolution` parameter (and a `bounds`/`extent` parameter at a later time)

Reply:
for 'admin_1_states_provinces', only one resolution is available (10m), so no choice here unfortunately